tests/components/suez_water/test_sensor.py

```diff
@@ -45,18 +48,27 @@ async def test_sensors_failed_update(
     assert mock_config_entry.state is ConfigEntryState.LOADED
 
     entity_ids = await hass.async_add_executor_job(hass.states.entity_ids)
-    assert len(entity_ids) == 1
-
-    state = hass.states.get(entity_ids[0])
-    assert entity_ids[0]
-    assert state.state != STATE_UNAVAILABLE
-
-    suez_client.fetch_aggregated_data.side_effect = PySuezError("Should fail to update")
+    assert len(entity_ids) == 2
+
+    for entity in entity_ids:
+        state = hass.states.get(entity)
+        assert entity
+        assert state.state != STATE_UNAVAILABLE
+
+    if method == "price":
+        suez_client.get_price.side_effect = PySuezError("Should fail to update")
+    elif method == "aggregated":
+        suez_client.fetch_aggregated_data.side_effect = PySuezError(
+            "Should fail to update"
+        )
+    else:
+        pytest.fail("Unknown update method")
```

Comment:
We don't want if statements in tests. You could use `getattr(suez_client, method).side_effect = ...` to do the same but without if statement

Reply:
Yeah I totally agree, I did look for the way to do exactly what you said but couldn't fint how. :smiley: